### PR TITLE
[Stats] Make metrics report time configurable

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This enables e2e test by allowing configuration of report time interval. It also adds a basic e2e test.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: src/ray/stats/stats.h [hunk 1]
```diff
@@ -89,6 +89,12 @@ static inline void Init(const TagsType &global_tags, const int metrics_agent_por
     exporter = exporter_to_use;
   }
 
+  // Set interval.
+  StatsConfig::instance().SetReportInterval(
+      absl::Seconds(RayConfig::instance().metrics_report_interval_seconds()));
+  StatsConfig::instance().SetHarvestInterval(
+      absl::Seconds(RayConfig::instance().metrics_report_interval_seconds() / 2));
```
Comment (simon-mo) [inline src/ray/stats/stats.h:96]: If `RayConfig::instance().metrics_report_interval_seconds() == 1` then `RayConfig::instance().metrics_report_interval_seconds() / 2 == 0` which is dangerously low. Maybe cap this to 1s with `std::max(RayConfig::instance().metrics_report_interval_seconds() / 2, 1)`?

File: src/ray/common/ray_config_def.h [hunk 2]
```diff
@@ -337,3 +337,6 @@ RAY_CONFIG(int64_t, gcs_server_request_timeout_seconds, 5)
 
 /// Whether to enable multi tenancy features.
 RAY_CONFIG(bool, enable_multi_tenancy, false)
+
+// The interval where metrics are exported in seconds.
+RAY_CONFIG(int64_t, metrics_report_interval_seconds, 10)
```
Comment (simon-mo) [inline src/ray/common/ray_config_def.h:342]: seconds should be usigned, uint64_t

**Review by simon-mo — APPROVED** [review]
Plz fix the edge cases with integer values in C++ before merge

File: src/ray/stats/stats.h (hunk 1 above)
Comment (rkooo567) [reply src/ray/stats/stats.h:96]: ah, really good catch. Will fix.

File: python/ray/tests/test_metrics_agent.py [hunk 3]
```diff
@@ -216,6 +217,70 @@ def get_metrics_export_address_from_node(nodes):
         loaded_json_data["targets"]))
 
 
+def test_metrics_export_end_to_end(ray_start_cluster):
+    NUM_NODES = 5
```
[the remark this answers is not in the document]
Comment (rkooo567) [reply python/ray/tests/test_metrics_agent.py:221]: reduced to 2

File: src/ray/common/ray_config_def.h (hunk 2 above)
Comment (rkooo567) [reply src/ray/common/ray_config_def.h]: Done.

**Comment by simon-mo** [comment]
I think you need to cast it the `1` to `static_cast<uint64_t>(1)`

```
ray/stats/stats.h:98:10: error: no matching function for call to 'max(uint64_t, int)'
         1));
```

**Comment by rkooo567** [comment]
@simon-mo Weird this wasn't caught in my local build.

**Comment by rkooo567** [comment]
Actually, I just made it milliseconds.

**Comment by simon-mo** [comment]
Failed test in CI 
https://travis-ci.com/github/ray-project/ray/jobs/371351775#L2285

**Comment by rkooo567** [comment]
@simon-mo @edoakes I pushed a fix. I will be surprised if it is still broken, but lmk if it does!

**Pull request merged**